ui-lib/src/androidTest/java/co/electriccoin/zcash/ui/common/ButtonTest.kt

```diff
@@ -0,0 +1,106 @@
+package co.electriccoin.zcash.ui.common
+
+import androidx.compose.foundation.interaction.MutableInteractionSource
+import androidx.compose.foundation.interaction.PressInteraction
+import androidx.compose.material3.Text
+import androidx.compose.ui.geometry.Offset
+import androidx.compose.ui.test.junit4.ComposeContentTestRule
+import androidx.compose.ui.test.junit4.createComposeRule
+import androidx.compose.ui.test.onNodeWithText
+import androidx.compose.ui.test.performClick
+import androidx.test.filters.MediumTest
+import cash.z.ecc.android.sdk.ext.collectWith
+import co.electriccoin.zcash.ui.design.theme.ZcashTheme
+import kotlinx.coroutines.CoroutineDispatcher
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.ExperimentalCoroutinesApi
+import kotlinx.coroutines.cancel
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.update
+import kotlinx.coroutines.launch
+import kotlinx.coroutines.test.StandardTestDispatcher
+import kotlinx.coroutines.test.advanceTimeBy
+import kotlinx.coroutines.test.resetMain
+import kotlinx.coroutines.test.runTest
+import kotlinx.coroutines.test.testTimeSource
+import org.junit.After
+import org.junit.Rule
+import org.junit.Test
+import kotlin.test.assertFalse
+import kotlin.test.assertTrue
+import kotlin.time.Duration
+import kotlin.time.Duration.Companion.seconds
+import kotlin.time.ExperimentalTime
+
+class ButtonTest {
+    @get:Rule
+    val composeTestRule = createComposeRule()
+
+    @OptIn(ExperimentalCoroutinesApi::class)
+    @After
+    fun tearDown() {
+        Dispatchers.resetMain()
+    }
+
+    @OptIn(ExperimentalTime::class, ExperimentalCoroutinesApi::class)
+    @Test
+    @MediumTest
+    fun timedButtonTest(): Unit = runTest {
+        val testDispatcher = StandardTestDispatcher(testScheduler)
+        val testSetup = newTestSetup(testDispatcher, 2.seconds)
+        val mark = testTimeSource.markNow()
+
+        testSetup.interactionSource.emit(PressInteraction.Press(Offset.Zero))
+        advanceTimeBy(3.seconds.inWholeMilliseconds)
+
+        launch {
+            testSetup.mutableActionExecuted.collectWith(this) {
+                if (!it) return@collectWith
+
+                assertTrue { mark.elapsedNow() >= 2.seconds }
+                this.cancel()
+            }
+        }
+    }
+
+    @OptIn(ExperimentalCoroutinesApi::class)
+    @Test
+    @MediumTest
+    fun buttonClickTest() = runTest {
+        val testDispatcher = StandardTestDispatcher(testScheduler)
+        val testSetup = newTestSetup(testDispatcher, 2.seconds)
+
+        composeTestRule.onNodeWithText("button").also {
+            it.performClick()
+        }
+        advanceTimeBy(3.seconds.inWholeMilliseconds)
+
+        launch {
```

Comment:
Can this be simplified down to just reading the value?

Reply:
without the launch block i can't get it to work. I'm always get After waiting for 60000 ms, the test coroutine is not completing. But when i try to cancel it like so : 
``` 
testSetup.mutableActionExecuted.collect {
    if (!it) return@collect
    assertTrue { mark.elapsedNow() >= 2.seconds }
    this.cancel()
} 
```
then i get TestScopeImpl was cancelled; job=TestScope[test ended]
so i thought using the launch was right approach

Reply:
Why can you not make this very very simple? `assertFalse(testSetup.mutableActionExecuted.value)`